AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -34,14 +34,97 @@ extension ApplePayComponent: PKPaymentAuthorizationViewControllerDelegate {
         paymentAuthorizationCompletion = completion
         let token = payment.token.paymentData.base64EncodedString()
         let network = payment.token.paymentMethod.network?.rawValue ?? ""
-        let billingContact = payment.billingContact
-        let shippingContact = payment.shippingContact
         let details = ApplePayDetails(paymentMethod: applePayPaymentMethod,
                                       token: token,
                                       network: network,
-                                      billingContact: billingContact,
-                                      shippingContact: shippingContact)
+                                      billingContact: payment.billingContact,
+                                      shippingContact: payment.shippingContact,
+                                      shippingMethod: payment.shippingMethod)
         
         submit(data: PaymentComponentData(paymentMethodDetails: details, amount: self.amountToPay, order: order))
     }
+
+    public func paymentAuthorizationViewController(
+        _ controller: PKPaymentAuthorizationViewController,
+        didSelectShippingContact contact: PKContact,
+        handler completion: @escaping (PKPaymentRequestShippingContactUpdate) -> Void
+    ) {
+        guard let applePayDelegate = applePayDelegate else {
+            return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))
+        }
+
+        let applePayPayment = applePayPayment
+        applePayDelegate.didUpdate(contact: contact,
+                                   for: applePayPayment,
+                                   completion: { [weak self] result in
```

Comment:
If I'm seeing correctly, all the completion for the 3 functions below are the same. Maybe we can move it into a function?

Reply:
`return completion(.init(paymentSummaryItems: applePayPayment.summaryItems))` returns different things.
I'll check if it is possible to make it generic

Reply:
I guess we can turn the if else block into a function then? if it makes sense